### ISSUE TITLE
Build a deterministic recent technology/development corpus from the local Markdown thread archive

### ISSUE THREAD

**Issue by RyderFreeman4Logos** [RyderFreeman4Logos/verbatim#312]
## Summary

Add a local corpus-builder that scans the configurable Markdown archive rooted by default at
`../drafts/articles/articles/`, selects recent technology and software-development threads, and
produces a deterministic private benchmark corpus manifest.

The archive contains thousands of recent political, economic, AI, and development-related
Twitter/X articles and threads. The initial benchmark subset should use technology/development
material so evaluations with Chinese SOTA models are not confounded by policy-sensitive civic or
geopolitical content.

## Motivation

Bible corpora are structurally convenient but likely represented in model training data. A
recent private archive offers fresher knowledge, internal document organization, noisy thread
structure, code and product identifiers, and facts that a model is less likely to know from
pretraining.

Filenames were generated by a local model and must not be treated as authoritative titles,
timestamps, topics, or gold labels. Selection and identity must come from file content and source
metadata.

## Corpus-builder behavior

- recursively scan Markdown files under a configured root;
- compute stable content hashes and source/thread identities;
- parse front matter and inline tweet/thread metadata when available;
- recover source URLs, post IDs, authors, timestamps, reply/quote relationships, and language;
- choose a deterministic date source with explicit fallback diagnostics;
- select a configurable recent window, default 183 days relative to a frozen cutoff date;
- classify into an allowlisted technology/development taxonomy;
- exclude politics, elections, geopolitics, civic controversy, and other policy-sensitive topics
  by default for the minimum-model suite;
- deduplicate exact and near-duplicate threads;
- split by source thread/document, never by chunk;
- preserve source Markdown unchanged and generate a separate manifest.

## Initial topic allowlist

```text
software-development
rust
systems
databases
search-and-retrieval
ai-ml-engineering
model-serving
hardware-and-accelerators
distributed-systems
security-engineering
web-and-mobile-development
developer-tools
open-source
data-infrastructure
```

The taxonomy is configurable and must support an `uncertain` quarantine.

## Scope

- [ ] Define a corpus-builder config and manifest schema.
- [ ] Add a dry-run inventory with inclusion/exclusion reason codes.
- [ ] Treat filenames as display hints only.
- [ ] Prefer source timestamps in content; record when filesystem time is the only fallback.
- [ ] Add deterministic topic rules and an optional bounded classifier review.
- [ ] Require independent validation before classifier output becomes a frozen corpus label.
- [ ] Add exact and near-duplicate detection.
- [ ] Preserve thread/reply/quote structure as metadata and optional deterministic graph edges.
- [ ] Detect files with missing, malformed, future, or contradictory timestamps.
- [ ] Produce private manifests containing hashes and local relative paths.
- [ ] Produce a safe public fixture with synthetic or redistributable samples.
- [ ] Never upload source text or filenames to a hosted model without explicit policy.
- [ ] Add rights, retention, and export checks.
- [ ] Make the root path and cutoff explicit; do not depend on the current working directory.

## Important evaluation semantics

This corpus measures retrieval and grounding against indexed sources. A tweet may be wrong about
the external world. Gold labels therefore distinguish:

- `source_supported`: the indexed source states the claim;
- `externally_verified`: optional, separately established real-world truth;
- `source_conflict`: indexed sources disagree;
- `unknown_truth`: no external verification was performed.

Verbatim must not conflate source-grounding with real-world truth.

## Acceptance criteria

- [ ] Repeated scans of unchanged files/config produce the same corpus manifest hash.
- [ ] The manifest records why every scanned file was included, excluded, or quarantined.
- [ ] A fixture proves that model-generated filenames do not control topic or title labels.
- [ ] Train/dev/test splits contain no shared source thread or near-duplicate document.
- [ ] The default minimum-model corpus contains only technology/development topics.
- [ ] No private article text is committed to the public repository.

## Dependencies and related work

- Builds on #263, #264, #270, #271, and #299.
- Supplies source material to `EVAL-012`, `EVAL-013`, and `FEEDBACK-001`.

---

## Addendum: Add grouped temporal splits and benchmark governance for recent Markdown threads

The recent technology/development corpus must avoid leakage from document families and untrusted metadata.

- [ ] Keep an entire tweet/thread, quoted-chain expansion, translation, mirror, and near-duplicate family in one split.
- [ ] Add a temporal holdout newer than the tuning corpus.
- [ ] Treat locally model-generated filenames as derived, low-confidence metadata; never use them as gold labels.
- [ ] Prefer body/front matter/source URL/timestamps with explicit provenance.
- [ ] Record corpus snapshot hash, grouping algorithm, filter rules, exclusions, and rights policy.
- [ ] Add opaque identifiers and document-internal recent facts to distinguish retrieval from model memory.
- [ ] Use no-context, oracle-context, and retrieved-context baselines.
- [ ] Govern benchmark edits through review, frozen final test data, confidence intervals, minimum effect sizes, and human sampling as specified by `EVAL-015`.
